projects/aws/upgrader/upgrade/syscalls.go

```diff
@@ -0,0 +1,56 @@
+//go:generate ../hack/tools/bin/mockgen -destination ./mocks/syscalls_mock.go -package mocks . SysCalls
+
+package upgrade
+
+import (
+	"context"
+	"io/fs"
+	"os"
+	"os/exec"
+)
+
+type SysCalls interface {
```

Comment:
Why is an interface needed? Are there multiple implementations for this? Also, this interface is too big.

Reply:
The idea of interfaces was to optimize for testability. We didn't want to make any actual system calls but also have a way to test the different upgrade flows. This was we could have the dependency injected and mocked out for out tests. I felt the size of interface was a reasonable trade-off given that these aren't super decoupled functions and all of them implicitly end up making os calls. I didn't feel the interface was necessarily bloated, but I will split it up in a more reasonable way if it adds value.